### PR TITLE
Updated Issuing elements to GA Version, Added functionality for card …

### PR DESCRIPTION
**Pull request by jpb-stripe**
…management, added functionality for Physical Cards

### CODE REVIEW

File: components/Stripe/CardStatusSwitchWidget.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          <div className="p-2 text-center ">
+            <p className="text-red-600">{errorText}</p>
+          </div>
+        ) : null}
+
+        <button
+          id="switch-card-status-btn"
+          type="button"
+          className="inline-flex items-center px-10 py-2.5 border border-transparent shadow-sm text-xs md:text-sm font-medium rounded-md text-white bg-accent-color hover:bg-accent-color-light ml-2 mr-2"
+          onClick={async (e) => {
```
Comment (cjavdev) [inline components/Stripe/CardStatusSwitchWidget.js:22]: lets move this handler above the JSX so it's easier to read.

File: pages/api/get_card.js [hunk 2]
```diff
@@ -1,42 +1,24 @@
...
-  }
-
-  const ephemeralKey = await getEphemeralKey(account, cardId, nonce);
 
+  const ephemeralKey = await stripe.ephemeralKeys.create({
+    nonce: nonce,
+    issuing_card: cardId,
+  }, {
+    stripeAccount: accountId,
+   apiVersion: '2020-03-02' });
```
Comment (cjavdev) [inline pages/api/get_card.js:16]: can we bump to the latest API version?

File: pages/api/issue_card.js [hunk 3]
```diff
@@ -15,20 +15,53 @@ export default async function handler(req, res) {
...
+        {
+          cardholder: req.body.cardholderid,
+          financial_account: financialAccount.id,
+          currency: 'usd',
+          shipping: {
+            name: cardholder.name,
+            service: 'standard',
+            type: 'individual',
+            address: {
+              line1: cardholder.billing.address.line1,
```
Comment (cjavdev) [inline pages/api/issue_card.js:36]: i think we can do some destructuring here or maybe pass address: cardholder.billing.address?

File: utils/format.js [hunk 4]
```diff
@@ -20,3 +20,7 @@ export function formatDateTime(secondsSinceEpoch) {
   };
   return d.toLocaleDateString('en-US', options);
 }
+
+export function capitalizeFirstLetter(string) {
```
Comment (cjavdev) [inline utils/format.js:24]: can we rename to `capitalize` to match other utility libraries like lodash.

Comment (cjavdev) [reply utils/format.js:24]: usually "uppercase" is used when making all letters capital, capitalize means make the first letter uppercase.

File: pages/api/switch_card_status.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+const stripe = require('stripe')(process.env.STRIPE_SECRET_KEY);
+
+export default async function handler(req, res) {
+  if (req.method === 'POST') {
+    const { app_auth } = parse(req.headers.cookie || '');
+    const session = decode(app_auth);
+    try {
+      const StripeAccountId = session.accountId;
+      const cardId = req.body.card_id;
+      let status = "";
```
Comment (cjavdev) [inline pages/api/switch_card_status.js:13]:
```js
const {new_status} = req.body;
let status = new_status == 'active' ? 'active' : 'inactive'
```

File: pages/api/issue_card.js [hunk 6]
```diff
@@ -15,20 +15,53 @@ export default async function handler(req, res) {
     });
 
     const financialAccount = financialAccounts.data[0];
+    const cardholder = await stripe.issuing.cardholders.retrieve(
+      req.body.cardholderid,
```
Comment (cjavdev) [inline pages/api/issue_card.js:19]: lets deconstruct the body params at the top so we can quickly see what this handler expects.

```js
const { cardholderid, card_type } = req.body
```

**Review by cjavdev — COMMENTED** [review]
thanks, juan!

File: components/Stripe/CardStatusSwitchWidget.js (hunk 1 above)
Comment (jpb-stripe) [reply components/Stripe/CardStatusSwitchWidget.js]: Fixed on commit [779b1b0](https://github.com/stripe-samples/treasury-nextjs/pull/19/commits/779b1b01fe53154e3ad6d07714f1c83c51607c17)

File: pages/api/get_card.js (hunk 2 above)
Comment (jpb-stripe) [reply pages/api/get_card.js]: Fixed on commit [779b1b0](https://github.com/stripe-samples/treasury-nextjs/pull/19/commits/779b1b01fe53154e3ad6d07714f1c83c51607c17)

File: pages/api/issue_card.js (hunk 3 above)
Comment (jpb-stripe) [reply pages/api/issue_card.js]: Fixed on commit [779b1b0](https://github.com/stripe-samples/treasury-nextjs/pull/19/commits/779b1b01fe53154e3ad6d07714f1c83c51607c17)

File: utils/format.js (hunk 4 above)
Comment (jpb-stripe) [reply utils/format.js]: Fixed on commit [779b1b0](https://github.com/stripe-samples/treasury-nextjs/pull/19/commits/779b1b01fe53154e3ad6d07714f1c83c51607c17)

File: pages/api/switch_card_status.js (hunk 5 above)
Comment (jpb-stripe) [reply pages/api/switch_card_status.js]: Fixed on commit [779b1b0](https://github.com/stripe-samples/treasury-nextjs/pull/19/commits/779b1b01fe53154e3ad6d07714f1c83c51607c17)

File: pages/api/issue_card.js (hunk 6 above)
Comment (jpb-stripe) [reply pages/api/issue_card.js]: Fixed on commit [779b1b0](https://github.com/stripe-samples/treasury-nextjs/pull/19/commits/779b1b01fe53154e3ad6d07714f1c83c51607c17)

**Review by cjavilla-stripe — APPROVED** [review]
lgtm

**Pull request merged**